scipy/signal/_waveforms.py

```diff
@@ -411,12 +440,25 @@ def chirp(t, f0, t1, f1, method='linear', phi=0, vertex_zero=True):
     >>> plot_spectrogram(f'Hyperbolic Chirp, f(0)=1500, f({T})=250', w, fs)
     >>> plt.show()
 
+    Complex linear chirp from -1000 Hz to 1000 Hz: 
+
+    >>> w = chirp(t, f0=-1000, f1=1000, t1=T, method="linear", complex=True)
+    >>> plot_spectrogram(f'Complex Linear Chirp, f(0)=-1000, f({T})=1000', w, fs, False)
+    >>> plt.show()
+
     """
     # 'phase' is computed in _chirp_phase, to make testing easier.
     phase = _chirp_phase(t, f0, t1, f1, method, vertex_zero)
     # Convert  phi to radians.
     phi *= pi / 180
-    return cos(phase + phi)
+
+    if complex:
+        a = exp(2j * pi * phi / (2*pi))
+        i = cos(phase + phi)
+        q = cos(phase + phi + (2 * pi * 90 / 360)) * 1j
+        return a * np.complex64(i - q)
+    else:
+        return cos(phase + phi)
```

Comment:
Is there a reason that you do incorporate [this](https://github.com/scipy/scipy/pull/17318#issuecomment-1311561161) or [this](https://github.com/scipy/scipy/pull/17318#issuecomment-1925889886) suggestions to simplify the code? I my opinion it would make the code much more readable.